internal/authorization/method_matcher.go

```diff
@@ -0,0 +1,19 @@
+package authorization
+
+import (
+	"github.com/authelia/authelia/internal/utils"
+)
+
+// isMethodMatching checks if the request method matches a method in the rule.
+func isMethodMatching(method string, methods []string) bool {
+	// If there is no regexp patterns, it means that we match any path.
+	if len(methods) == 0 {
+		return true
+	}
+
+	if method == "" {
```

Comment:
What's this case meant to be for?
Should we not handle this in the validator on start-up?

Reply:
This handles if the header is empty technically. So if the check has for example:
```go
isMethodMatching("", []string{"GET", "POST"})
```